src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -923,6 +956,738 @@ private FindRequestContext ConstructFindRequestContext(RestRequestContext contex
             return dbResultSetRow;
         }
 
+        /// <summary>
+        /// Performs the given GraphQL create mutation operation.
+        /// </summary>
+        /// <param name="entityName">Name of the top level entity</param>
+        /// <param name="parameters">Mutation parameter arguments</param>
+        /// <param name="sqlMetadataProvider">SqlMetadaprovider</param>
+        /// <param name="context">Hotchocolate's context for the graphQL request.</param>
+        /// <param name="multipleInputType">Boolean indicating whether the create operation is for multiple items.</param>
+        /// <returns>Primary keys of the created records (in the top level entity).</returns>
+        /// <exception cref="DataApiBuilderException"></exception>
+        private List<IDictionary<string, object?>> PerformMultipleCreateOperation(
+                string entityName,
+                IMiddlewareContext context,
+                IDictionary<string, object?> parameters,
+                ISqlMetadataProvider sqlMetadataProvider,
+                bool multipleInputType = false)
+        {
+            string fieldName = multipleInputType ? MULTIPLE_INPUT_ARGUEMENT_NAME : SINGLE_INPUT_ARGUEMENT_NAME;
+
+            // Parse the hotchocolate input parameters into .net object types
+            object? parsedInputParams = GQLMultipleCreateArgumentToDictParams(context, fieldName, parameters);
+
+            if (parsedInputParams is null)
+            {
+                throw new DataApiBuilderException(message: "The create mutation body cannot be null.",
+                                                  statusCode: HttpStatusCode.BadRequest,
+                                                  subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
+            }
+
+            // List of Primary keys of the created records in the top level entity.
+            // Each dictionary in the list corresponds to the PKs of a single record.
```

Comment:
```suggestion
            // Each dictionary in the list corresponds to the PK of a single record. Will contain only one entry in case of a point multiple-create mutation.
```